iocore/net/UnixNetVConnection.cc

```diff
@@ -622,8 +617,7 @@ UnixNetVConnection::get_data(int id, void *data)
 VIO *
 UnixNetVConnection::do_io_read(Continuation *c, int64_t nbytes, MIOBuffer *buf)
 {
-  ink_assert(c || 0 == nbytes);
```

Comment:
Maybe move this assert after the new check and add an equivalent one to ``do_io_write``?